### PR TITLE
support foundry sign() cheatcode

### PR DESCRIPTION
**Pull request by cd1m0**
What title sayas.

### CODE REVIEW

File: test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	    // Correct address
+	    address addr = 0x16bB6031CBF3a12B899aB99D96B64b7bbD719705;
+	    // Incorrect address
+	    address badAddr = 0x16bb6031CbF3a12B899AB99d96b64b7BBD719706;
+
+	    (uint8 v, bytes32 r, bytes32 s) = vm.sign(privKey, digest);
+
+	    address rec = ecrecover(digest, v, r, s);
+
+	    assertEq(rec, badAddr);
```
Comment (blitz-1306) [inline test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol:23]: Variable says `badAddr`. Why do we need `addr` then? What do we actually expect here?

**Review by blitz-1306 — COMMENTED** [review]
Great. Just one question.

File: test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol (hunk 1 above)
Comment (blitz-1306) [reply test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol:23]: Just figured that we intentionally want test to fail. In that case - things are fine. Still, I would be grateful if ther would be a comment, that explicitly state our intentions (describe the case in words, if possible).

**Review by blitz-1306 — APPROVED** [review]

File: test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol (hunk 1 above)
Comment (cd1m0) [reply test/samples/local/foundry_sign_addr/contracts/test/Cheatcodes.t.sol:23]: Sorry, I have to re-do the whole campaign for the one comment, so for now will skip. Hopefully we won't have to look at this code again

**Pull request merged**